images/handlers.go

```diff
@@ -182,65 +183,60 @@ func SetChildrenLabels(manager content.Manager, f HandlerFunc) HandlerFunc {
 	}
 }
 
-// FilterPlatformList is a handler wrapper which limits the descriptors returned
-// by a handler to the specified platforms.
-func FilterPlatformList(f HandlerFunc, platformList ...string) HandlerFunc {
+// FilterPlatforms is a handler wrapper which limits the descriptors returned
+// based on matching the specified platform matcher.
+func FilterPlatforms(f HandlerFunc, m platforms.Matcher) HandlerFunc {
 	return func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
 		children, err := f(ctx, desc)
 		if err != nil {
 			return children, err
 		}
 
-		if len(platformList) == 0 {
-			return children, nil
-		}
-
-		var m platforms.Matcher
+		var descs []ocispec.Descriptor
 
-		if len(platformList) > 0 {
-			ps := make([]ocispec.Platform, len(platformList))
-			for i, platform := range platformList {
-				p, err := platforms.Parse(platform)
-				if err != nil {
-					return nil, err
+		if m == nil {
+			descs = children
+		} else {
+			for _, d := range children {
+				if d.Platform == nil || m.Match(*d.Platform) {
+					descs = append(descs, d)
 				}
-				ps[i] = p
-			}
-			m = platforms.Any(ps...)
-		}
-
-		var descs []ocispec.Descriptor
-		for _, d := range children {
-			if d.Platform == nil || m.Match(*d.Platform) {
-				descs = append(descs, d)
 			}
 		}
 
 		return descs, nil
 	}
 }
 
-// FilterPlatforms is a handler wrapper which limits the descriptors returned
-// based on matching the specified platform matcher.
-func FilterPlatforms(f HandlerFunc, m platforms.Matcher) HandlerFunc {
+// LimitManifests is a handler wrapper which filters the manifest descriptors
+// returned using the provided platform, sorting and limiting the number
+// of results to the provided limit.
+// A limit of 0 or less is considered no limit.
```

Comment:
Include a statement here about favoring entries that appear first in the manifests.